lib/perl/Genome/InstrumentData/Command/Import/WorkFlow/FastqsToBam.pm

```diff
@@ -57,6 +61,42 @@ sub execute {
     return 1;
 }
 
+sub _unarchive_fastqs_if_necessary {
+    my $self = shift;
+    $self->debug_message('Unarchive fastqs if necessary...');
+
+    my @new_fastq_paths;
+    for my $fastq_path ( $self->fastq_paths ) {
+        if ( $fastq_path !~ /\.gz$/ ) {
+            $self->debug_message('Unarchive not necessary for '.$fastq_path);
+            push @new_fastq_paths, $fastq_path;
+            next;
+        }
+        $self->debug_message('Unarchiving: '.$fastq_path);
+        try {
+            Genome::Sys->shellcmd(cmd => "gunzip $fastq_path");
+        }
+        catch {
+            $self->error_message($_) if $_;
+            $self->error_message('Failed to gunzip fastq!');
+            return;
```

Comment:
That's more of a preference choice.

In both examples I've noticed you could do:

```
my $success = try {
    # same try block
} 
catch {
    # same catch block
}
return unless $success;
```